var/spack/repos/builtin/packages/mysql/package.py

```diff
@@ -105,8 +119,34 @@ def cmake_args(self):
             options.append('-DWITHOUT_SERVER:BOOL=ON')
         return options
 
+    def fix_dtrace_usr_bin_path(self, spack_env):
+        # dtrace may cause build to fail because it uses
+        # '/usr/bin/python' in the shebang. To work around that we copy
+        # the original script into a temporary folder, and change the
+        # shebang to '/usr/bin/env python'.
+        dtrace = which('dtrace').path
+        dtrace_copy_path = os.path.join(tempfile.mkdtemp(), 'dtrace-copy')
+        dtrace_copy = os.path.join(dtrace_copy_path, 'dtrace')
+        mkdirp(dtrace_copy_path)
+        copy(dtrace, dtrace_copy)
+        filter_file(
+            '^#!/usr/bin/python',
+            '#!/usr/bin/env python',
+            dtrace_copy
+        )
+        # To have our own copy of dtrace in PATH, we need to
+        # prepend to PATH the temporary folder where it resides.
+        spack_env.prepend_path('PATH', dtrace_copy_path)
+
     def setup_environment(self, spack_env, run_env):
         cxxstd = self.spec.variants['cxxstd'].value
         flag = getattr(self.compiler, 'cxx{0}_flag'.format(cxxstd))
         if flag:
             spack_env.append_flags('CXXFLAGS', flag)
+        if cxxstd > 11:
+            spack_env.append_flags('CXXFLAGS', '-Wno-deprecated-declarations')
```

Comment:
Will this clause run if `cxxstd=98`????  Also, what is the type of `self.spec.variants['cxxstd'].value`?  (My guess is Python `str`, so we shouldn't be comparing it to an `int`).  We need to work a little harder on making sure types and ordering work out here.

Reply:
@citibeth _Mea culpa_. Please see https://github.com/spack/spack/pull/11356/commits/0acc9d30fc7c877394f6357b1dad057ada3ec551 for a proposed fix.